app/assets/javascripts/govuk_publishing_components/components/checkboxes.js

```diff
@@ -33,7 +33,8 @@ window.GOVUK.Modules.Checkboxes = window.GOVUKFrontend;
     if (window.GOVUK.analytics && window.GOVUK.analytics.trackEvent) {
       // Where checkboxes are manipulated externally in finders, `suppressAnalytics`
       // is passed to prevent duplicate GA events.
-      if (!event.detail || (event.detail && event.detail.suppressAnalytics !== true)) {
+      var allowAnalytics = ((event || {}).detail || {}).suppressAnalytics !== true
```

Comment:
I'm finding this hard to parse - would it be better to split this over several lines? Or into several variables?

Reply:
I'll have a look, see what I can do.

Reply:
This is a very clever technique I found online for finding whether a nested attribute exists in an object. It looks complicated at first, but it's actually quite simple. Here's an example.

Say you want to check for the existence of a nested attribute `object.first.second`, but `first` might not exist. If you try `if (object.first.second)` the code will error, because `first` doesn't exist. Using this technique, at each level of the evaluation, the code returns either the attribute it finds (e.g. `first`) or an empty object. That then gets appended to the next level down, which means an error can't occur, we get to the end of the expression and either find what we're looking for, or we find a series of empty nested objects doesn't contain what we're looking for.

Breaking this up into more lines of code was what I was trying to avoid as it gets really verbose. I was nesting multiple if statements (`if (event) { if (event.detail) { if (event.detail.suppressAnalytics) { }}}`).

I've added a comment in the code to explain this and provided a link to more details.

Reply:
Thanks - the link will definitely help future me who'll completely forget about this and go "huh" when i read this again 😄  👍 

Can't wait to be able to use optional chaining!